dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/constraints/DatadogDataConstraints.kt

```diff
@@ -67,6 +68,34 @@ internal class DatadogDataConstraints : DataConstraints {
         return convertedAttributes.take(MAX_ATTR_COUNT).toMap()
     }
 
+    override fun validateEvent(rumEvent: Any): Any {
+        return if (rumEvent is ViewEvent) {
+
+            val customTimings = rumEvent.view.customTimings?.let {
+                it.copy(
+                    additionalProperties = it.additionalProperties.mapKeys { entry ->
+                        val sanitizedKey =
+                            entry.key.replace(Regex("[^a-zA-Z0-9\\-_.@$]"), "_")
```

Comment:
I think something similar with this RegEx is being used below in the `tagTransfomers`. Can we extract this in a static const ?

Reply:
they are different, `tagTransforms` defines different set of allowed characters

Reply:
Ok...nevermind then 👍 .